### DISCUSSION TITLE
How to time client request duration

### DISCUSSION THREAD

**Discussion by andrewdavey** [ravendb/ravendb#12432] [Q&A] [answered]
What's the best way to time how long requests made by the RavenDB client are taking?

I see that the `DocumentStore` has `OnBeforeRequest` and `OnSucceedRequest` events. Do these capture all types of requests made by the client? My idea is to record the start time in the event args `Request.Properties` then use that when the request succeeds.

Or is there a simpler/better way to access request timing stats?

**Comment by ppekrol** [comment, marked as answer]
That should work, yes. But you also need to take into account that request can fail, there is a `OnFailedRequest` event available as well.

**Reply by andrewdavey** [reply]
OK, but the args for `OnFailedRequest` do not have a `Request` property.

**Reply by ppekrol** [reply]
I think this is something that could be added if needed.

**Reply by andrewdavey** [reply]
Thanks. That might useful at some point. For now I'm happy just to get a basic measure of successful requests.

**Reply by ppekrol** [reply]
I've created an issue in our YT. It is available here: https://issues.hibernatingrhinos.com/issue/RavenDB-16936
